core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -96,7 +95,6 @@ class Replica(val brokerId: Int, val topicPartition: TopicPartition) extends Log
     */
   private def updateLastSentHighWatermark(highWatermark: Long): Unit = {
     _lastSentHighWatermark = highWatermark
-    trace(s"Updated HW of replica to $highWatermark")
```

Comment:
I don't think this trace call is worthwhile given the cost. It would be hard to turn it on without being deluged with logs..

Reply:
@junrao Thoughts? Can we remove these calls?

Reply:
At minimum I think we could keep the trace I deleted here: https://github.com/apache/kafka/pull/8468#discussion_r407832642 which should end up logging the lastSentWatermark anyway?

Reply:
Yes, I agree this may not be that useful. We have other logging like the following in Partition.updateFollowerFetchState() that provides similar traceability. 

```
        debug(s"Recorded replica $followerId log end offset (LEO) position " +
          s"${followerFetchOffsetMetadata.messageOffset} and log start offset $followerStartOffset.")

```